samples: Add esp32s2_saola overlay and remove esp32_temp_sensor

hirokin94 opened the pull request:
- Add overlay for the esp32s2_saola board to die_temp_polling sample.
- Add aliases for the die_temp_polling sample to esp32s2 dtsi.
- Remove esp32_temp_sensor.

Signed-off-by: Hiroki Tada tada.hiroki@fujitsu.com

soburi reviewed (commented):
Could you split the commit into "Adding esp32s2_saola to die_temp_polling" and "Removing "esp32_temp_sensor sample"?

hirokin94 committed:
samples: Add esp32s2_saola overlay

- Add overlay for the esp32s2_saola board to die_temp_polling sample.
- Add aliases for the die_temp_polling sample to esp32s2 dtsi.

Signed-off-by: Hiroki Tada <tada.hiroki@fujitsu.com>

hirokin94 committed:
samples: Remove esp32_temp_sensor

- esp32_temp_sensor is removed as die_temp_polling is used instead of it.

Signed-off-by: Hiroki Tada <tada.hiroki@fujitsu.com>

marekmatej commented:
@hirokin94 
May I ask what is the reason for removing the `esp32_temp_sensor`?
Also, FYI there is no SOC `ESP32-C2` neither board named `esp32c2_saola`

sylvioalves commented on samples/sensor/die_temp_polling/boards/esp32s2_saola.overlay:
```diff
@@ -1,5 +1,5 @@
 /*
- * Copyright (c) 2022 Espressif Systems (Shanghai) Co., Ltd.
```
Any reason to remove current Copyright?

hirokin94 replied:
In the past commit, it was pointed out that the copyright should be change, so I changed it this commit as well.
https://github.com/zephyrproject-rtos/zephyr/pull/55426#discussion_r1125418191

marekmatej replied:
IMO That was a different situation since you created a new file, unlike the current case.

hirokin94 commented:
> @hirokin94 May I ask what is the reason for removing the `esp32_temp_sensor` sample? What is the replacement for that sample? Also, FYI there is no such SOC as `ESP32-C2` nor board named `esp32c2_saola`

- May I ask what is the reason for removing the esp32_temp_sensor sample? What is the replacement for that sample?
`esp32_temp_sensor` can be completely replaced to `die_temp_polling` samples.
In the following commit, stm32-specific sample were replaced `die_temp_polling` sample.
https://github.com/zephyrproject-rtos/zephyr/pull/56803
Same for esp32, I think that esp32_temp_sensor should be merged into die_temp_polling.

- Also, FYI there is no such SOC as ESP32-C2 nor board named 
Thank you for pointing out. I fixed it.

MaureenHelm commented on samples/sensor/die_temp_polling/boards/esp32s2_saola.overlay:
```diff
@@ -0,0 +1,11 @@
+ */
+
+&coretemp {
+    status = "okay";
```
Replace spaces with a tab

hirokin94 replied:
Fixed it.

MaureenHelm reviewed (changes requested):
The twister failure doesn't look related to these changes. I've been out for a while and don't know if it's been fixed in main already, but please rebase to pick up any fixes that may have been merged.

MaureenHelm reviewed (approved):
Code changes look good, but still need to resolve the failed CI checks

hirokin94 commented:
All CI checks were passed.
